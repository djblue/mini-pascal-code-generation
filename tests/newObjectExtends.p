program main;

class boop
begin
  var aa : array[1..5] of integer;
end


class beep extends boop
begin
  var yy, xx : integer;
end

class main 
begin

  var bb : beep;
      parent : boop;
      ii : integer;

  function main;
  begin

    bb := new beep;
    bb.xx := 56;
    print bb.xx;
    bb.yy := 24;
    print bb.yy;

    ii := 0;
    while  ii < 6 do
      begin
        bb.aa[ii] := ii;
        ii := ii + 1
      end;

    ii := 5;
    while ii > 0 do
      begin
        print bb.aa[ii];
        ii := ii - 1
      end

  end
end
.
