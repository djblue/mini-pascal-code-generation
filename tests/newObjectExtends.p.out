56
24
5
4
3
2
1
